pkg/util/helper/workstatus.go

```diff
@@ -84,22 +85,34 @@ func assembleWorkStatus(c client.Client, selector labels.Selector, workload *uns
 			klog.Errorf("Failed to get manifestIndex of workload in work.Spec.Workload.Manifests. Error: %v.", err)
 			return nil, err
 		}
+		clusterName, err := names.GetClusterName(work.Namespace)
+		if err != nil {
+			klog.Errorf("Failed to get clusterName from work namespace %s. Error: %v.", work.Namespace, err)
+			return nil, err
+		}
+
+		// if sync work to member cluster failed, then set status back to resource binding.
+		for _, c := range work.Status.Conditions {
+			if c.Status == v1.ConditionFalse {
+				aggregatedStatus := workv1alpha1.AggregatedStatusItem{
```

Comment:
It is better to to check the type of condition although we just use 'WorkApplied' type now.